test/base_test_fixture.h

```diff
@@ -1012,6 +1013,24 @@ struct base_test_fixture
         REQUIRE(!connection.dbms_version().empty());
     }
 
+    void get_info_test()
+    {
+        // A generic test to exercise the DBMS info API is callable.
+        // DBMS-specific test (MySQL, SQLite, etc.) may perform extended checks.
+        nanodbc::connection connection = connect();
+        REQUIRE(!connection.get_info<nanodbc::string_type>(SQL_DRIVER_NAME).empty());
+        REQUIRE(!connection.get_info<nanodbc::string_type>(SQL_ODBC_VER).empty());
+
+       // Test SQLUSMALLINT resutls
+        REQUIRE(connection.get_info<unsigned short>(SQL_ODBC_INTERFACE_CONFORMANCE) > 0);
```

Comment:
Querying for `SQL_ODBC_INTERFACE_CONFORMANCE` returns `SQLUINTEGER` (4 byte).

So, `unsigned short` is too narrow type and `SQLGetInfo` filling 4 into 2 bytes `T value` will corrupt the stack.